source/dofs/dof_renumbering.cc

```diff
@@ -2248,6 +2254,187 @@ namespace DoFRenumbering
            ExcInternalError());
   }
 
+
+
+  template <int dim, int spacedim>
+  void
+  support_point_wise(DoFHandler<dim, spacedim> &dof_handler)
+  {
+    std::vector<types::global_dof_index> renumbering(
+      dof_handler.n_locally_owned_dofs(), numbers::invalid_dof_index);
+    compute_support_point_wise(renumbering, dof_handler);
+
+    // If there is only one component then there is nothing to do, so check
+    // first:
+    if (Utilities::MPI::max(renumbering.size(),
+                            dof_handler.get_communicator()) > 0)
+      dof_handler.renumber_dofs(renumbering);
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  compute_support_point_wise(
+    std::vector<types::global_dof_index> &new_dof_indices,
+    const DoFHandler<dim, spacedim> &     dof_handler)
+  {
+    const types::global_dof_index n_dofs = dof_handler.n_locally_owned_dofs();
+    Assert(new_dof_indices.size() == n_dofs,
+           ExcDimensionMismatch(new_dof_indices.size(), n_dofs));
+
+    // This renumbering requires that we first do the component renumbering.
+    // We'll start there and then compose the renumberings.
+    std::vector<types::global_dof_index> component_renumbering(
+      n_dofs, numbers::invalid_dof_index);
+    compute_component_wise<dim, spacedim>(component_renumbering,
+                                          dof_handler.begin_active(),
+                                          dof_handler.end(),
+                                          std::vector<unsigned int>(),
+                                          false);
+    // Same logic as in compute_component_wise(): if there is only one component
+    // we have nothing to do
+    if (component_renumbering.size() == 0)
+      {
+        new_dof_indices.resize(0);
+        return;
+      }
+    std::fill(new_dof_indices.begin(),
+              new_dof_indices.end(),
+              numbers::invalid_dof_index);
+
+    // This is now equal to what dof_handler.locally_owned_dofs() would be if we
+    // executed the renumbering
+    IndexSet component_renumbered_dofs(dof_handler.n_dofs());
+    component_renumbered_dofs.add_indices(component_renumbering.begin(),
+                                          component_renumbering.end());
+
+    // Determine the cellwise renumbering for each element:
+    std::map<unsigned int, std::vector<unsigned int>> cell_renumberings;
+    unsigned int                                      fe_index = 0;
+    for (const FiniteElement<dim, spacedim> &fe :
+         dof_handler.get_fe_collection())
```

Comment:
Not that it makes a big difference, but why don't you use a vector for the cell renumberings and push back into that vector?